linera-views/src/common.rs

```diff
@@ -153,9 +153,11 @@ pub trait KeyValueStoreClient {
         let Some(value) = value else {
             return Ok(None);
         };
-        let Some(first_value) = *value.first() else {
+        let first_value = value.first();
+        if first_value.is_none() {
             return Err(DatabaseConsistencyError::ZeroLengthValue.into());
-        };
+        }
+        let first_value = *first_value.unwrap();
```

Comment:
Did the `if let` not compile?

Reply:
It does not.

Reply:
Why? Maybe I just forgot that it's a reference? Actually, we don't even need `if let`:

```rust
        let first_value = *value.first().ok_or(DatabaseConsistencyError::ZeroLengthValue)?;
```

(Not a blocker, I'm just eager to have as few `unwrap`s and `expect`s in production code as possible.)

I'm also wondering whether we should use "byte" instead of "value" here. We already have two different `Value` types in the system. :dizzy_face: